File: .leetcode/14.最长公共前缀.cpp
/*
 * @lc app=leetcode.cn id=14 lang=cpp
 *
 * [14] 最长公共前缀
 */
#include<bits/stdc++.h>
using namespace std;

// @lc code=start
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string same = "";
        if(!strs.size())return same;
        same = strs[0];
        if(strs.size()==1)return same;
        int n = strs.size();
        for(int i = 1;i < n;i++){
            int length = min(same.size(),strs[i].size());
            int j = 0;
            for(j;j < length;j++){
                if(same[j] != strs[i][j]){
                    break;
                }
            }
            same = same.substr(0,j);
        }
        return same;
    }
};
// @lc code=end

File: .leetcode/14.最长公共前缀.java
/*
 * @lc app=leetcode.cn id=14 lang=java
 *
 * [14] 最长公共前缀
 */

// @lc code=start
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs.length == 0){
            return "";
        }
        String ans = strs[0];
        int n = strs.length;
        for(int i = 1;i < n;i++){
            int j;
            for(j = 0;j < ans.length();j++){
                if(j >= strs[i].length()||ans.charAt(j) != strs[i].charAt(j)){
                    break;
                }
            }
            ans = ans.substring(0,j);
        }
        return ans;
    }
}
// @lc code=end

